File: 백준/Bronze/1934. 최소공배수/최소공배수.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let input = [];
rl.on('line', (line) => {
    input.push(line);
});

rl.on('close', () => {
    let T = parseInt(input[0]);
    
    for (let i = 1; i <= T; i++) {
        let [A, B] = input[i].split(' ').map(Number);
        
        let lcm = getLCM(A, B);
        console.log(lcm);
    }
});

// 최대공약수를 구하는 함수 (유클리드 호제법 사용)
function getGCD(a, b) {
    while (b !== 0) {
        let temp = b;
        b = a % b;
        a = temp;
    }
    return a;
}

// 최소공배수를 구하는 함수
function getLCM(a, b) {
    return a * (b / getGCD(a, b));
}

File: 백준/Bronze/1934. 최소공배수/최소공배수.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int t = scanner.nextInt();

        for(int i = 0; i < t; i++) {
            int a = scanner.nextInt();
            int b = scanner.nextInt();

            int lcm = lcm(a, b);
            System.out.println(lcm);
        }

        scanner.close();
    }

    public static int gcd(int a, int b) {
        while (b != 0) {
            int temp = b;
            b = a % b;
            a = temp;
        }

        return a;
    }

    public static int lcm(int a, int b) {
        return (a * b) / gcd(a, b);
    }

}
